narwhals/_arrow/namespace.py

```diff
@@ -437,6 +437,14 @@ def __call__(self: Self, df: ArrowDataFrame) -> Sequence[ArrowSeries]:
         condition = parse_into_expr(self._condition, namespace=plx)(df)[0]
         try:
             value_series = parse_into_expr(self._then_value, namespace=plx)(df)[0]
+            if len(value_series) == 1:  # literal case
+                value_series = condition.__class__._from_iterable(
+                    pa.repeat(pa.scalar(value_series[0]), len(condition)),
+                    name="literal",
+                    backend_version=self._backend_version,
+                    version=self._version,
+                )
```

Comment:
does it work to use `broadcast_series` like we do below?

Reply:
I think it would look like:

```py
if len(value_series) == 1:  # literal case
    _, value_series_native = broadcast_series(
        [condition, value_series]
    )
    value_series = condition.__class__(
        value_series_native,
        name="literal",
        backend_version=self._backend_version,
        version=self._version,
    )
```
Not sure how much cleaner this is 🙃